File: online/karleksbrev/attachments_source/karleksbrev/python3/karleksbrev.py
#!/usr/bin/python3
def encode(S: str) -> str:
    E = []
    for c in S:
        E.append(c*5)
    return "".join(E)

# Note: this E is E after it has been corrupted
def decode(E: str) -> str:
    ret = ""
    while len(E):
        chunk = E[:5]
        E = E[5:]
        ret += chunk[2]
    return ret

File: online/karleksbrev/attachments_source/karleksbrev/rust/karleksbrev.rs
pub fn encode(S: String) -> String {
    let mut E = String::new();
    for c in S.chars() {
        E.push_str(&c.to_string().repeat(5));
    }
    E
}

// Note: this E is E after it has been corrupted
pub fn decode(E: String) -> String {
    let mut ret = String::new();
    let mut chars = E.chars();
    while let Some(chunk) = chars.by_ref().take(5).collect::<String>().chars().nth(2) {
        ret.push(chunk);
    }
    
    ret
}
